src/components/utils/popover.ts

```diff
@@ -5,39 +5,7 @@ import SearchInput from './search-input';
 import EventsDispatcher from './events';
```

Comment:
This file is way too much large, maybe we can split it somehow to follow SOLID?

Reply:
there's a plan to refactor popover in further pull requests. probably it will even be moved to separate repository. I wouldn't touch it for now